package/MDAnalysis/analysis/pca.py

```diff
@@ -0,0 +1,234 @@
+        # for transform function
+        self._demean = demean
+        # access 0th index
+        self._u.trajectory[0]
+        # reference will be 0th index
+        self._reference = self._u.atoms.select_atoms(select)
+        self._atoms = atomgroup.select_atoms(select)
+        self.n_components = n_components
+        self._n_atoms = self._atoms.n_atoms
+        self._calculated = False
+
+    def _prepare(self):
+        n_dim = self._n_atoms * 3
+        self.cov = np.zeros((n_dim, n_dim))
+        if self._demean:
+            self._ref_atoms = self._reference
+            self._ref_cog = self._ref_atoms.center_of_geometry()
+
+    def _single_frame(self):
+        if self._demean:
+            # TODO make this more functional, initial proof of concept
+            mobile_atoms, old_rmsd = _fit_to(self._atoms.positions,
+                                             self._ref_atoms.positions,
+                                             self._atoms, self._ref_cog,
+                                             self._atoms.center_of_geometry())
+            # now all structures are aligned to reference
+            x = mobile_atoms.positions.ravel()
+        else:
+            x = self._atoms.positions.ravel()
+        self.cov += np.dot(x[:, np.newaxis], x[:, np.newaxis].T)
+
+    def _conclude(self):
+        self.cov /= self.n_frames - 1
+        e_vals, e_vects = np.linalg.eig(self.cov)
+        sort_idx = np.argsort(e_vals)[::-1]
+        self.variance = e_vals[sort_idx]
+        self.p_components = e_vects[sort_idx]
+        self.cumulated_variance = (np.cumsum(self.variance) /
+                                   np.sum(self.variance))
+        self._calculated = True
+
+    def transform(self, atomgroup, n_components=None, start=None, stop=None,
+                  step=None):
+        """Apply the dimensionality reduction on a trajectory
+
+        Parameters
+        ----------
+        atomgroup: MDAnalysis atomgroup
+            The atoms to be PCA transformed.
+        n_components: int, optional
+            The number of components to be projected onto, Default none: maps
+            onto all components.
+        start: int, optional
+            The frame to start on for the PCA transform. Default: None becomes
+            0, the first frame index.
+        stop: int, optional
+            Frame index to stop PCA transform. Default: None becomes n_frames.
+            Iteration stops *before* this frame number, which means that the
+            trajectory would be read until the end.
+        step: int, optional
+            Number of frames to skip over for PCA transform. Default: None
+            becomes 1.
+        Returns
+        -------
+        pca_space : array, shape (number of frames, number of components)
+        """
+        if not self._calculated:
+            self.run()
+        # TODO has to accept universe or trajectory slice here
+        if self._atoms != atomgroup:
```

Comment:
this test is to strict.  See cell 9 of [this notebook](https://gist.github.com/kain88-de/237a13d53b4999d08fd53d68a732377b). If I have two atomgroups with the same selection the warning shouldn't be triggered.